### DISCUSSION TITLE
SDK5

### DISCUSSION THREAD

**Discussion by uwewoessner** [RosettaDrone/rosettadrone#157] [General]
Hello everyone,
how are your thought about supporting SDK5.x and DJI Mini3?

**Comment by kripper** [comment]
Sure, but must be added in parallel to SDK v4, since not all models are supported by SDK v5.
This will requiere an option to select the desired SDK version during runtime in the ConnectionActivity.
Should be developed in a separate branch and merged once stable.

**Comment by uwewoessner** [comment]
Dear Christopher,
I will check if I can find a student to work on it. Can you estimate how much work this is?

**Comment by kripper** [comment]
Sorry  I'm not familiar with SDK 5 differences.

**Reply by m4xw** [reply]
Quite a lot changed internally but should be fairly easy to port, was on my todo eventually

**Reply by m4xw** [reply]
Having a SDK bridge will need a abstraction/middleware tho, might as well start direct mavlink support too if we do that

**Comment by adpmdrones** [comment]
We've compiled RosettaDrone with DJI SDK v5 but iot seems there's an error with Google Play Services.
Any suggestion?
Thanks in advance
![main_screen](https://github.com/RosettaDrone/rosettadrone/assets/4736951/c8baadab-614d-45b5-8a4e-4a51c0e88203)
![20231219_074024](https://github.com/RosettaDrone/rosettadrone/assets/4736951/3dcace78-06a4-495c-8ff9-97de98440a8a)

**Reply by heejun0328** [reply]
Can you share what specific changes have you made for SDK V5?

**Reply by adpmdrones** [reply]
We're still working on the SDK V4 - V5 porting.

**Reply by heejun0328** [reply]
Great to hear!! I am also trying to working on it.
since I am very new to android app applications, so can you advice me how to start?
Maybe starting to look at MSDK v5 API reference and translate?

**Reply by adpmdrones** [reply]
Check the differences between SDK v4 and v5. They're quite different
